analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -723,29 +728,61 @@ def add_arguments_to_parser(parser):
         func=main, func_process_config_file=cmd_config.process_config_file)
 
 
-def __get_skip_handler(args) -> skiplist_handler.SkipListHandler:
+def get_source_file_paths(
+    file_filters: List[str],
+    compile_commands: tu_collector.CompilationDB,
+    header_file_extensions=(
+        '.h', '.hh', '.H', '.hp', '.hxx', '.hpp', '.HPP', '.h++', '.tcc')
+) -> List[str]:
     """
-    Initialize and return a skiplist handler if
-    there is a skip list file in the arguments or files options is provided.
+    Returns a list of source files for existing header file otherwise returns
+    with the same file path expression.
     """
-    skip_file_content = ""
+    file_paths = []
+    for file_filter in file_filters:
+        file_paths.append(file_filter)
```

Comment:
The goal of the function would be clearer if its result wouldn't include these header files. The function's name is `get_source_file_paths()` where "source" indicates that non-headers are returned. At least we are using "source file" for non-headers conventionally throughout in our naming conventions. The list of headers could be concatenated to the source files at the caller.

I'm not sure, but I'm thinking if this function is necessary at all. Couldn't `tu_collector.get_dependent_sources()` be used directly? As far as I can see, the added behavior of this function is to check only existing headers in the given list. What is the reason of this restriction? The result of this function goes to a skipfile where it is not a big problem if the file doesn't exist.

Reply:
I renamed this function to `get_affected_file_paths`.